proposals/0000-swift-settings.md

```diff
@@ -0,0 +1,446 @@
+compiler flags to be enabled on a single file. Initially this will only be used
+to support controlling the default isolation. For example:
+
+```swift
+#SwiftSettings(
+    .defaultIsolation(MainActor.self)
+)
+```
+
+In the future, we wish to extend this to also include support for controlling
+`warningsAsErrors`, `strictConcurrency`, and other flags. See future directions
+for more details.
+
+## Motivation
+
+Today Swift contains multiple ways of changing compiler and language behavior
+from the command line including specifying the default actor isolation via
+[Controlling Default Actor Isolation](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0466-control-default-actor-isolation.md). Since
+these compiler and language behaviors can only be manipulated via the command
+line and the compiler compiles one module at a time, users are forced to apply
+these flags one module at a time instead of one file at a time. When confronted
+with this, users are forced to split the subset of files into a separate module
+creating an unnecessary module split. We foresee that this will occur more often
+in the future if new features like [Controlling Default Actor Isolation](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0466-control-default-actor-isolation.md)
+are accepted by Swift Evolution. Consider the following situations where
+unnecessary module splitting would be forced:
+
+* Consider a framework that by default uses `nonisolated` isolation, but wants
+  to expose a set of UI elements for users of the framework. In such a case, it
+  is reasonable to expect the author to want to make those specific files
+  containing UI elements to have a default isolation of `@MainActor` instead of
+  nonisolated. Without this feature one must introduce an additional module to
+  contain the UI elements for the framework when semantically there is really
+  only one framework.
+
+* Imagine an app that has enabled `@MainActor` isolation by default since it
+  contains mostly UI elements but that wishes to implement helper abstractions
+  that interact with a database on a background task. This helper code is
+  naturally expressed as having a default isolation of nonisolated implying that
+  one must either split the database helpers into a separate module or must mark
+  most declarations in the database code with nonisolated by hand.
+
+## Proposed solution
+
+We propose introducing a new macro called `#SwiftSettings` that takes a list of
+structs of type `SwiftSetting`. `SwiftSetting` will possess a static method for
+each command line argument that it supports. Any parameters that would be passed
+on the command line are passed as parameters to the static method. The compiler
+upon parsing a `#SwiftSettings` macro updates its internal state to set the
+appropriate settings before compiling the rest of the file. An example of such a
+`#SwiftSettings` invocation is the following:
```

Comment:
I didn't see a mention in the detailed design about the reason why you chose this spelling, which is to match the Swift Package Manifest API so there's a consistent spelling for settings in source code. I recommend putting that either here or in the detailed design.